protocol/lavasession/consumer_session_manager.go

```diff
@@ -85,6 +85,26 @@ func (csm *ConsumerSessionManager) UpdateAllProviders(epoch uint64, pairingList
 	return nil
 }
 
+func (csm *ConsumerSessionManager) UpdateMaxCULimit(virtualEpoch, prevVirtualEpoch uint64) {
+	for key, consumerSessionWithProvider := range csm.pairing {
+		if consumerSessionWithProvider == nil {
+			continue
+		}
+
+		consumerSessionWithProvider.MaxComputeUnits = consumerSessionWithProvider.MaxComputeUnits / (prevVirtualEpoch + 1) * (virtualEpoch + 1)
+		csm.pairing[key] = consumerSessionWithProvider
+	}
+
+	for key, consumerSessionWithProvider := range csm.pairingPurge {
```

Comment:
pairingPurge are providers from the previous epoch, they do not need a CU increase as the epoch has passed. 
so no need for this section